maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/FileMatcher.java

```diff
@@ -62,6 +62,36 @@ static List<FileMatcher> create(String pattern) {
         return List.of(new ContainsMatcher(pattern));
     }
 
+    /**
+     * Create matcher from git pattern.
+     *
+     * @param pattern pattern to parse
+     * @return matcher that can be matched against a {@link io.helidon.build.maven.enforcer.FileRequest}
+     */
+    static FileMatcher createFromGitPattern(String pattern) {
```

Comment:
Maybe use an enum `PatternFormat` `DEFAULT|GITIGNORE` and have a create method that takes an enum value and a string ; that method that delegate to sub-methods like `createDefault` and `createGitIgnore`.